### PR TITLE
Adding FastViT

### PR DESCRIPTION
**Pull request by veb-101**
Added the base model block.

### CODE REVIEW

**Comment by veb-101** [comment]
* Fixed issue in `MobileOneBlock` to handle a case when `out_channels = k * in_channels`.
* The above fix introduced another issue during the re-parameterization between depthwise conv. and batch-norm weights. The fix was to reshape the batch norm weights to have a similar shape to the depthwise weights. Instead of simple `(1, 1, -1, 1)`, it's now `(1, 1, -1. depth_multiplier)`.
* Added `ReparamLargeKernelConv` block. The block code is largely similar to `MobileOneBlock`
